src/Language/Rust.ts

```diff
@@ -0,0 +1,312 @@
+import { TargetLanguage } from "../TargetLanguage";
+import { TypeGraph } from "../TypeGraph";
+import { ConvenienceRenderer, ForbiddenWordsInfo } from "../ConvenienceRenderer";
+import {
+    legalizeCharacters, splitIntoWords, isLetterOrUnderscoreOrDigit, combineWords,
+    isLetterOrUnderscore, allLowerWordStyle, firstUpperWordStyle, intToHex, utf32ConcatMap, escapeNonPrintableMapper, isPrintable, isAscii, isLetterOrDigit
+} from "../Strings";
+import { Name, Namer, funPrefixNamer } from "../Naming";
+import { UnionType, nullableFromUnion, Type, ClassType, matchType, removeNullFromUnion, EnumType } from "../Type";
+import { Sourcelike, maybeAnnotated } from "../Source";
+import { anyTypeIssueAnnotation, nullTypeIssueAnnotation } from "../Annotation";
+
+export default class RustTargetLanguage extends TargetLanguage {
+
+    protected get rendererClass(): new (graph: TypeGraph, ...optionValues: any[]) => ConvenienceRenderer {
+        return RustRenderer;
+    }
+
+    constructor() {
+        super("Rust", ["rs", "rust", "rustlang"], "rs");
+        this.setOptions([]);
+    }
+}
+
+const keywords = [
+    // Special reserved identifiers used internally for elided lifetimes,
+    // unnamed method parameters, crate root module, error recovery etc.
+    "{{root}}",
+    "$crate",
+
+    // Keywords used in the language.
+    "as",
+    "box",
+    "break",
+    "const",
+    "continue",
+    "crate",
+    "else",
+    "enum",
+    "extern",
+    "false",
+    "fn",
+    "for",
+    "if",
+    "impl",
+    "in",
+    "let",
+    "loop",
+    "match",
+    "mod",
+    "move",
+    "mut",
+    "pub",
+    "ref",
+    "return",
+    "self",
+    "Self",
+    "static",
+    "struct",
+    "super",
+    "trait",
+    "true",
+    "type",
+    "unsafe",
+    "use",
+    "where",
+    "while",
+
+    // Keywords reserved for future use.
+    "abstract",
+    "alignof",
+    "become",
+    "do",
+    "final",
+    "macro",
+    "offsetof",
+    "override",
+    "priv",
+    "proc",
+    "pure",
+    "sizeof",
+    "typeof",
+    "unsized",
+    "virtual",
+    "yield",
+
+    // Weak keywords, have special meaning only in specific contexts.
+    "catch",
+    "default",
+    "dyn",
+    "'static",
+    "union",
+];
+
+const isAsciiLetterOrDigit = (codePoint: number, index: number): boolean => {
+    if (!isAscii(codePoint)) {
+        return false;
+    }
+
+    return index === 0
+        ? isLetterOrDigit(codePoint)
+        : isLetterOrUnderscoreOrDigit(codePoint);
+};
+
+const legalizeName = legalizeCharacters(isAsciiLetterOrDigit);
+
+function snakeCaseStyle(original: string): string {
+    const words = splitIntoWords(original);
+    return combineWords(
+        words,
+        legalizeName,
+        allLowerWordStyle,
+        allLowerWordStyle,
+        allLowerWordStyle,
+        allLowerWordStyle,
+        "_",
+        isLetterOrUnderscore

```

Comment:
`combineWords` can handle start characters, which I think you shouldn't need the index for the start character in `legalizeName`.  But maybe I'm missing something here.

Reply:
I have removed the index parameter and limited the first char to a letter. How can I limit the first character to be a:

* letter or underscore if length(name) > 1
* letter if length(name) == 1

Reply:
Nevermind, I have added a new commit fixing this issue.